test/parallel/test-async-hooks-promise-triggerid.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const async_hooks = require('async_hooks');
+
+const promiseAsyncIds = [];
+
+async_hooks.createHook({
+  init: common.mustCallAtLeast((id, type, triggerId) => {
+    if (type === 'PROMISE') {
+      assert.strictEqual(promiseAsyncIds[promiseAsyncIds.length - 1] || 1,
```

Comment:
I found the assert hard to read, so either add a comment or remove it. You should definitely check that `async_hooks.triggerId()` is correct, since that is the second parameter in `push_ids`.